### PR TITLE
Add V2 support

### PR DESCRIPTION
**Pull request by sdlyy**
Resolves L2B-3930
Resolves L2B-3531

- Added chain configurations
- Added naming consts
- Split addresses into logical groups
- Added simple V1/V2 switch - V2 currently disabled on staging
- Added base, simple components for V2s

### CODE REVIEW

File: packages/backend/src/peripherals/clients/ProviderCache.ts [hunk 1]
```diff
@@ -14,14 +14,14 @@ export class ProviderCache implements DiscoveryCache {
...
-    chainId: number,
+    chainId: string,
     blockNumber: number,
   ): Promise<void> {
     await this.repository.addOrUpdate({
       key,
       value,
-      chainId: ChainId(chainId),
-      blockNumber: blockNumber,
+      chainId: ChainId(Number(chainId)),
```
Comment (michalsidzej) [inline packages/backend/src/peripherals/clients/ProviderCache.ts:23]: This is dirty, but fine for now

File: packages/frontend/src/hooks/useDiscoveryApi.ts [hunk 2]
```diff
@@ -48,6 +48,7 @@ export function useDiscoveryApi({
         setIsError(false)
         setIsLoading(false)
       } catch (e) {
+        console.error(e)
```
Comment (michalsidzej) [inline packages/frontend/src/hooks/useDiscoveryApi.ts:51]: clog (can stay here if needed, just pointing out)

File: packages/frontend/src/view/components/BlockchainAddress.tsx [hunk 3]
```diff
@@ -85,7 +87,12 @@ export function BlockchainAddress(props: Props) {
             )}
           >
             <span className="sm:hidden">{addressEllipsis(props.address)}</span>
-            <span className="hidden sm:inline"> {props.address}</span>
+            <span className="hidden sm:inline">
+              {' '}
+              {props.alwaysShort
+                ? addressEllipsis(props.address)
+                : props.address}
+            </span>
```
Comment (michalsidzej) [inline packages/frontend/src/view/components/BlockchainAddress.tsx:95]:
```suggestion
            <span className="hidden sm:inline">
              {props.alwaysShort
                ? addressEllipsis(props.address)
                : props.address}
            </span>
```

**Review by michalsidzej — COMMENTED** [review]
Well done. One problem I see with running both discoveries at the same time, we're increasing the amount of unnecessary calls. This is not crutial to solve rn, but can become a problem in the future. Overall LGTM.

**Comment by sdlyy** [comment]
@michalsidzej 
> Well done. One problem I see with running both discoveries at the same time, we're increasing the amount of unnecessary calls. This is not crutial to solve rn, but can become a problem in the future. Overall LGTM.

Yeah, I will resolve discovery config optimization as a part of separate PR since this beefy guys locks me out from easy task split

**Review by michalsidzej — APPROVED** [review]

**Pull request merged**